timemachine/cpp/src/context.cu

```diff
@@ -71,7 +71,14 @@ double Context::_get_temperature() {
 
 void Context::setup_local_md(double temperature, bool freeze_reference) {
     if (this->local_md_pots_ != nullptr) {
-        throw std::runtime_error("local md already configured");
+        if (this->local_md_pots_->temperature != temperature ||
+            this->local_md_pots_->freeze_reference != freeze_reference) {
+            throw std::runtime_error(
+                "local md configured with different parameters, current parameters: Temperature " +
+                std::to_string(this->local_md_pots_->temperature) + " Freeze Reference " +
+                std::to_string(this->local_md_pots_->freeze_reference));
+        }
+        return;
```

Comment:
To be clear, would calling `setup_local_md` multiple times with different parameters lead to an inconsistent state, or is this intended to catch bugs in application code (assuming that any usage where we call this multiple times with different parameters is invalid)?

Reply:
We could change out the underlying `LocalPotentials` object if there were different parameters, just risk paying the cost of setting up the potentials. This was mostly intended to catch application level bugs, as there is not currently a clear use case for different types of local MD configurations on a single context.